### PR TITLE
Pedersen libfunc

### PR DESCRIPTION
**Pull request by edg-l**
Implements the pedersen builtin type and the hash libfunc.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -45,6 +45,7 @@ tracing-subscriber = { version = "0.3.17", features = [
 
 [dev-dependencies]
 cairo-lang-runner = "2.0.1"
+cairo-felt = "0.6.1"
```
Comment (azteca1998) [inline Cargo.toml:48]:
```suggestion
cairo-felt = "0.6.1"
cairo-lang-runner = "2.0.1"
```

File: src/libfuncs/felt252.rs [hunk 2]
```diff
@@ -331,7 +331,7 @@ mod test {
     }
 
     // Parse numeric string into felt, wrapping negatives around the prime modulo.
-    fn f(value: &str) -> [u32; 8] {
+    pub fn f(value: &str) -> [u32; 8] {
```
Comment (azteca1998) [inline src/libfuncs/felt252.rs:334]: I think we should move it to utils where all the shared testing stuff is.

File: src/libfuncs/pedersen.rs [hunk 3]
```diff
@@ -1,3 +1,272 @@
...
+    TType: GenericType,
+    TLibfunc: GenericLibfunc,
+    <TType as GenericType>::Concrete: TypeBuilder<TType, TLibfunc, Error = CoreTypeBuilderError>,
+    <TLibfunc as GenericLibfunc>::Concrete: LibfuncBuilder<TType, TLibfunc, Error = Error>,
+{
+    {
+        metadata
+            .get_mut::<RuntimeBindingsMeta>()
+            .expect("Runtime library not available.");
+    }
```
Comment (azteca1998) [inline src/libfuncs/pedersen.rs:77]: Is there any reason why you have it within a block?

**Review by azteca1998 — APPROVED** [review]
Nice work.

File: src/libfuncs/pedersen.rs (hunk 3 above)
Comment (edg-l) [reply src/libfuncs/pedersen.rs:77]: yes, lifetime issues, metadata is borrowed mutable somewhere

Comment (edg-l) [reply src/libfuncs/pedersen.rs:77]: ok never mind the issue is no longer there

**Review by igaray — APPROVED** [review]

**Pull request merged**